lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -33,20 +33,25 @@ void circt::firrtl::emitConnect(ImplicitLocOpBuilder &builder, Value dst,
   auto dstType = dstFType.dyn_cast<FIRRTLBaseType>();
   auto srcType = srcFType.dyn_cast<FIRRTLBaseType>();
 
-  // Special Connects
-  if (!dstType) {
-    if (dstFType.isa<RefType>() &&
-        !dstFType.cast<RefType>().getType().hasUninferredWidth())
-      builder.create<RefConnectOp>(dst, src);
-    else // Other types, give up and leave a connect
-      builder.create<ConnectOp>(dst, src);
+  // References use ref.assign.  Types should match, leave to verifier if not.
+  if (isa<RefType>(dstFType)) {
+    builder.create<RefDefineOp>(dst, src);
     return;
   }
 
   // If the types are the exact same we can just connect them.
-  // Strict connect does not allow uninferred widths.
-  if (dstType == srcType && !dstType.hasUninferredWidth()) {
-    builder.create<StrictConnectOp>(dst, src);
+  if (dstFType == srcFType) {
```

Comment:
Won't this now strictconnnect out-of-dialect types (e.g. firesim)?

Reply:
This preserves the previous behavior, if this was intended to change I didn't expect that as part of adding RefConnect which this PR starts by reverting for clarityand review-/rebase-friendliness since it's main value was splitting out references from connections which it only did partially and ref.define handles more explicitly while modelling reference behavior re:single-connect/working-from-under-when.  FWIW this is why I rushed to intercept last Friday before it landed (see #4803, which wasn't ready but was trying to avoid the need for the conflict).

FWIW the only actual change to emitConnect made here is adding the use of `ref.define` if they're reference types, see this commit:  https://github.com/llvm/circt/pull/4812/commits/fa53310a0f9da9e674af2eff0cd6763f19373aba .

If we want to change how this works, let's do it explicitly and perhaps add a test :+1: (although not sure if we have a preferred way to test foreign types?).

Since you've pointed this out, how about we file an issue that we want to change this and we'll get to it? I'm happy to take that on re:implementation.

Thinking a bit more, previously / currently "strictconnect" just means the connection types are the same (is it mentioned/documented/intended to mean something different?), so I'm not sure why it's inappropriate for foreign types as long as they match (the check here).

Reply:
Well, reverted this to the pre-PR behavior other than using ref define if destination is a reference.